drinkfood/app/Http/Controllers/Public/CartController.php

```diff
@@ -109,6 +128,97 @@ private function createOrderDetail($req, $uidOrder)
         }  
     }
 
+    private function calcTotalOrder($arrayOrder)
+    {
+        $totalOrder = 0;
+        for($i = 0; $i < count($arrayOrder); $i++)
+        {
+            $quantityProduct = $arrayOrder[$i][0]['price'] * count($arrayOrder[$i]);
+            $totalOrder += $quantityProduct;
+        }
+        return number_format($totalOrder, 0, ',', '.')." VND";
+    }
+
+    public function removeProduct(Request $req)
+    {
+        $deleteProduct = OrderDetail::where([['id_product','=',$req->idProduct], ['id_order','=',$req->idOrder]])->delete();
+        if($deleteProduct == true)
+        {
+            /* Delete session product saved */
+            $req->session()->forget($req->uidProduct);
+            return ['status'=>true, 'message'=>__('message.delete_product_order_success')];
+        }else return ['status'=>false, 'message'=>__('message.delete_product_order_error')];
+    }
+
+    public function setSession(Request $request)
+    {
+        $arrayInfoProduct = ['uid'=>$request->uidProduct, 'id'=>$request->idProduct, 'idorder'=>$request->idOrder, 'quantity'=>$request->quantity, 'amount'=>$request->amountProduct];
```

Comment:
xem lại request nào 
https://laravel.com/docs/8.x/requests#retrieving-a-portion-of-the-input-data